### PR TITLE
Add C/C++ demo files

### PR DESCRIPTION
**Pull request by mcaveety**
@SharpRake here is the associated PR for the C/C++ getting started guide. This should contain all of the files needed to follow the guide. :)

### CODE REVIEW

File: c/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
+# C/C++ Demo
+
+This repository contains a simple demonstration of C and C++ programs using Chainguard Images. For a complete tutorial, please refer to the Chainguard Academy [Getting Started with the C/C++ Chainguard Images guide](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/c/).
```
Comment (SharpRake) [inline c/README.md:3]:
```suggestion
This repository contains two demo applications for C and C++, respectively, both of which use Chainguard Images. For a complete tutorial, please refer to the Chainguard Academy [Getting Started with the C/C++ Chainguard Images guide](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/c/).
```

File: c/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+```sh
+docker build -f Dockerfile1 -t example1:latest .
+```
+
+Once your image build it complete, run the following command to start a container.
+```sh
+docker run --name example1 example1:latest
+```
+
+You should see output in your terminal identical to that of the binary we compiled locally.
```
Comment (SharpRake) [inline c/README.md:48]:
```suggestion
You will see output in your terminal identical to that of the binary we compiled locally.
```

**Review by SharpRake — DISMISSED** [review]
This looks good to me! I don't know if those Wiz warnings are important - if you haven't already maybe double check with @erikaheidi that it's kosher if they're there?

File: c/README.md (hunk 1 above)
Comment (mcaveety) [inline c/README.md:3]:
```suggestion
This repository contains three examples showing single and multi-stage builds for C-based programs using Chainguard Images. For a complete tutorial, please refer to the Chainguard Academy [Getting Started with the C/C++ Chainguard Images guide](https://edu.chainguard.dev/chainguard/chainguard-images/getting-started/c/).
```

File: c/README.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+You should see the following output in your terminal.
+
+```Output
+Hello, world!
+I am a demo from the Chainguard Academy.
+My code was written in C.
+```
+
+Now, with Docker Engine running, we can compile this program inside of the `gcc-glibc` Chainguard Image. We will use `Dockerfile1` for this image build.
```
Comment (mcaveety) [inline c/README.md:29]:
```suggestion
Now, with Docker Engine running, you can compile this program inside of the `gcc-glibc` Chainguard Image. You will use `Dockerfile1` for this image build.
```

File: c/README.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+```sh
+docker build -f Dockerfile1 -t example1:latest .
+```
+
+Once your image build it complete, run the following command to start a container.
+```sh
+docker run --name example1 example1:latest
+```
+
+You will see output in your terminal identical to that of the binary we compiled locally.
```
Comment (mcaveety) [inline c/README.md:48]:
```suggestion
You will see output in your terminal identical to that of the binary you compiled locally.
```

**Comment by mcaveety** [comment]
So, I checked on this with Erika and Adrian, it looks like some of the warnings can be ignored. The ones with high severity occur because there is no user specified in the Dockerfile, but it doesn't look like `gcc-glibc` has a non-root user? I'm piecing this together from what I understand... I'm getting the impression that we can push this anyway, Adrian was mentioning filing an issue about it so if that happens I suppose I can update further

... I did the thing again where I dismissed your review. I'm not sure exactly what is causing that 🥲 I am so sorry!

**Comment by mcaveety** [comment]
Update 8/12: Some changes are being made to the named users of the `gcc-glibc` image. Because of this, changes to the written guide need to be made. This will be reopened when these changes are ready to be reviewed.

**Comment by mcaveety** [comment]
I have fun and wonderful news! @SharpRake we've made some changes so that the images run as a non-root user, user `65532`. This should hopefully resolve the Wiz scanner complaints. I updated the readme to reflect these changes. this should be ready to go, and the guide will follow shortly!

**Review by SharpRake — APPROVED** [review]
approving again, thanks Michelle!

**Pull request merged**